zpi-frontend/src/pages/Theses/ThesisStudent.tsx

```diff
@@ -0,0 +1,261 @@
+          const response = await api.put(api_access + 'reservation/' + reservation.id,
+            JSON.stringify(reservation)
+          );
+
+          if (response.status === 200) {
+            toast.success(t('thesis.readyForApproval'));
+            console.log('All users reservations sent for approval successfully');
+          }
+        } catch (error) {
+          toast.error(t('thesis.readyForApprovalError'));
+          console.error(`Failed to update reservations for reservation: ${reservation}`, error);
+        }
+      }
+    }
+  };
+
+  const downloadDeclaration = () => {
+    let url = api_access + 'report/pdf/thesis-declaration/' + thesis?.id;
+
+    let toastId: any = null;
+    toastId = toast.info(t('thesis.generating'), { autoClose: false });
+
+    api.get(url, { responseType: 'blob' })
+      .then((response) => {
+        const file = new Blob([response.data], { type: 'application/pdf' });
+        const fileURL = URL.createObjectURL(file);
+        const link = document.createElement('a');
+        link.href = fileURL;
+
+        const contentDisposition = response.headers['content-disposition'];
+        let filename = 'report.pdf';
+        if (contentDisposition) {
+          const filenameMatch = contentDisposition.match(/filename=(.+)/i);
+          if (filenameMatch.length === 2)
+            filename = filenameMatch[1];
+        }
+        link.setAttribute('download', filename);
+        document.body.appendChild(link);
+        link.click();
+
+        setTimeout(() => {
+          toast.dismiss(toastId);
+        }, 2000);
+        toast.success(t('thesis.downloadSuccessful'));
+      })
+
+      .catch((error) => {
+        console.error(error);
+        setTimeout(() => {
+          toast.dismiss(toastId);
+        }, 2000);
+
+        if (error.response.status === 401 || error.response.status === 403) {
+          setAuth({ ...auth, reasonOfLogout: 'token_expired' });
+          handleSignOut(navigate);
+        }
+        else if (error.response.status === 404) {
+          toast.error(t('thesis.downloadNoDataError'));
+        }
+        else
+          toast.error(t('thesis.downloadError'));
+      });
+  }
+
+  return (
+    <div className='page-margin'>
+      {loaded ? (<React.Fragment>
+        <div className='d-flex justify-content-between align-items-center'>
+          <div>
+            {(thesis && thesis.status.name === "PendingApproval") &&
+              <Alert variant="info">
+                {t('studentThesis.pendingApproval')}
+              </Alert>
+            }
+            {(thesis && thesis.reservations.every(r => r.readyForApproval)) &&
+              (thesis.reservations.some(r => r.confirmedBySupervisor) ?
+                <Alert variant="info">
+                  {t('studentThesis.afterSupervisorAccept')}
+                </Alert>
+                :
+                <Alert variant="info">
+                  {t('studentThesis.beforeSupervisorAccept')}
+                </Alert>)
```

Comment:
Fajnie by było jeszcze dodać alert o tym, że rezerwacja czeka na potwierdzenie od lidera i że jest już potwierdzona przez lidera. A liderowi alert o tym, że są jakieś rezerwacje do rozpatrzenia. Jeszcze bym konsekwentnie sprawdzała wszystkie rezerwacje czy są zaakceptowane przez prowadzącego lub tylko tę konkretną czy jest ready for approval